packages/mongo-nav/src/MongoNav.tsx

```diff
@@ -111,27 +136,55 @@ export default function MongoNav({
   urls: urlsProp,
   constructOrganizationURL: constructOrganizationURLProp,
   constructProjectURL: constructProjectURLProp,
+  onError = () => {},
+  onSuccess = () => {},
 }: MongoNavInterface) {
-  let data;
+  const [data, setData] = React.useState<DataInterface | undefined>(undefined);
 
-  if (mode === Mode.Dev) {
-    data = devModeData;
+  const hosts = defaultsDeep(hostsProp, defaultHosts);
+  const endpointURI = `${hosts.cloud}/user/shared`;
+
```

Comment:
Up to you but you could move all this fetching related behavior to a custom hook.

Reply:
Talked offline, leaving as-is for now